res/controllers/common-hid-packet-parser.js

```diff
@@ -393,17 +393,16 @@ class HIDPacket {
             console.error(`HIDPacket.pack - Parsing packed value: invalid pack format ${field.pack}`);
             return;
         }
-        const bytes = this.packSizes[field.pack];
-        const signed = this.signedPackFormats.includes(field.pack);
         if (field.type === "bitvector") {
             const bitVector = /** @type {HIDBitVector} */ (field.value);
-            if (bytes > 1) {
+            if (this.packSizes[field.pack] > 1) {
                 console.error("HIDPacket.pack - Packing multibyte bit vectors not yet supported");
                 return;
             }
-            for (const bit_id in bitVector.bits) {
-                const bit = bitVector.bits[bit_id];
-                data[field.offset] = data[field.offset] | bit.value;
+            const bitVectorKeyArr = Object.keys(bitVector.bits);
+            let bitVectorIdx = bitVectorKeyArr.length;
+            while (bitVectorIdx--) {
+                data[field.offset] |= bitVector.bits[bitVectorKeyArr[bitVectorIdx]].value;
```

Comment:
In Qt it would be an issue that issue that the original loop is over a non const, which performs a deep copy.  
Did you test the following? In Qt it would be equal fast compared to the proposed solution.   
```suggestion
            const bitVectorKeyArr = Object.keys(bitVector.bits);
            for (const bit in bitVectorKeyArr) {
                data[field.offset] |= bit.value;
```

Reply:
`const` in js is not the same as `const` in C++. The underlying datastructure is still mutable when declared via `const`, the only difference is that the variable can't be rebound later. Either way, I don't think the usual Qt wisdom applies here.

Reply:
The proposed code is no functional equivalent. I modified it as follows to benchmark it against the other versions:
```suggestion
            const bitVectorKeyArr = Object.keys(bitVector.bits);
            for (const bit of bitVectorKeyArr) {
                data[field.offset] |= bitVector.bits[bit].value;
```

Reply:
Can you drop a comment about your results to the code to avoid discussing this a gain?